### ISSUE TITLE
Can not resolve hbase

### ISSUE THREAD

**Issue by hujianxin** [alexarchambault/coursier#325]
The error message is:

```
[error] 
[error] 1 error(s):
[error]   org.apache.hbase:${compat.module}:1.1.5:
[error]     not found: /home/hujianxin/.sbt/0.13/plugins/target/resolution-cache/org.apache.hbase/${compat.module}/1.1.5/resolved.xml.xml
[error]     Not found
[error]     not found: /home/hujianxin/.ivy2/local/org.apache.hbase/${compat.module}/1.1.5/ivys/ivy.xml
[error]     not found: https://repo1.maven.org/maven2/org/apache/hbase/$%7Bcompat.module%7D/1.1.5/$%7Bcompat.module%7D-1.1.5.pom
[error]     not found: /home/hujianxin/.m2/repository/org/apache/hbase/$%7Bcompat.module%7D/1.1.5/$%7Bcompat.module%7D-1.1.5.pom
[trace] Stack trace suppressed: run last *:coursierResolution for the full output.
[error] (*:coursierResolution) java.lang.Exception: Encountered 1 error(s) in dependency resolution
[error] Total time: 2 s, completed Aug 15, 2016 8:49:33 AM
```

**Comment by hujianxin** [comment]
I ran it in `1.0.0-M13` and `v1.0.0-M12-1`, got the same error.

**Comment by hujianxin** [comment]
But it works on hbase 1.2.2, weird

**Comment by alexarchambault** [comment]
I quickly tried to reproduce that, without success. Do you have more details? Like the dependency that transitively pulled hbase?

**Comment by longhua** [comment]
I got a similar error when I used coursier with the following configuration. However, it worked if I removed coursier. Is there any suggestion? Thanks in advance.

```
libraryDependencies ++= Seq(
  "org.apache.spark" %% "spark-streaming" % "1.6.2" % "provided",
  "org.apache.spark" %% "spark-sql" % "1.6.2" % "provided",
  "org.apache.spark" %% "spark-streaming-kafka" % "1.6.2" exclude("org.spark-project.spark", "unused"),
  "org.elasticsearch" %% "elasticsearch-spark" % "2.1.2" intransitive(),
  "org.apache.phoenix" % "phoenix-spark" % "4.7.0-HBase-1.1" % "provided",
  "com.typesafe" % "config" % "1.3.0",
  "com.github.nscala-time" %% "nscala-time" % "2.6.0",
  "joda-time" % "joda-time" % "2.9.1"
)
```

```
coursier.ResolutionException: Encountered 2 error(s) in dependency resolution:
  org.apache.hbase:${compat.module}:1.1.3:
    not found:
      /Users/lhuang/.ivy2/local/org.apache.hbase/${compat.module}/1.1.3/ivys/ivy.xml
      https://repo1.maven.org/maven2/org/apache/hbase/$%7Bcompat.module%7D/1.1.3/$%7Bcompat.module%7D-1.1.3.pom
  sqlline:sqlline:1.1.8:
    not found:
      /Users/lhuang/.ivy2/local/sqlline/sqlline/1.1.8/ivys/ivy.xml
      https://repo1.maven.org/maven2/sqlline/sqlline/1.1.8/sqlline-1.1.8.pom
[error] (*:coursierResolution) coursier.ResolutionException: Encountered 2 error(s) in dependency resolution:
[error]   org.apache.hbase:${compat.module}:1.1.3:
[error]     not found:
[error]       /Users/lhuang/.ivy2/local/org.apache.hbase/${compat.module}/1.1.3/ivys/ivy.xml
[error]       https://repo1.maven.org/maven2/org/apache/hbase/$%7Bcompat.module%7D/1.1.3/$%7Bcompat.module%7D-1.1.3.pom
[error]   sqlline:sqlline:1.1.8:
[error]     not found:
[error]       /Users/lhuang/.ivy2/local/sqlline/sqlline/1.1.8/ivys/ivy.xml
[error]       https://repo1.maven.org/maven2/sqlline/sqlline/1.1.8/sqlline-1.1.8.pom
[error] Total time: 17 s, completed Aug 29, 2016 11:51:13 PM
```

**Comment by xsy233** [comment]
I got a similar issue.
It seems coursier cannot parse ${compat.module} correctly
my workaround: 

"org.apache.hbase" % "hbase-server" % hbaseV
  exclude("org.apache.hbase", "${compat.module}")

**Comment by longhua** [comment]
@xsy233 Thanks a lot. It works like a charm.

**Comment by alexarchambault** [comment]
This seems to originate from the lack of support in coursier for profile activations like
```xml
<activation>
  <property>
    <name>!hadoop.profile</name>
  </property>
</activation>
```
that is, activation expecting a property not to be defined I guess (via the `!`).

That shouldn't be hard to fix.

**Comment by eboto** [comment]
The just-linked PR tests and fixes the specific profile activation behavior that @alexarchambault identified. If that is indeed the root cause of this issue, then the issue can probably be closed after the PR merges.

**Comment by TopSpoofer** [comment]
@eboto  when?

**Comment by eboto** [comment]
Assuming CI passes, then whenever @alexarchambault has a chance to look at it =)

Afterwards, I assume a new build will be baked after [these issues](https://github.com/alexarchambault/coursier/milestone/7) are resolved

**Comment by alexarchambault** [comment]
This should be fixed in `1.0.0-M14-6`.

**Comment by mantovani** [comment]
> I got a similar issue.
> It seems coursier cannot parse ${compat.module} correctly
> my workaround:
> 
> "org.apache.hbase" % "hbase-server" % hbaseV
> exclude("org.apache.hbase", "${compat.module}")

thank you, just worked!
